format_spec/tile.md

```diff
@@ -0,0 +1,253 @@
+# Tile
+
+## Main Structure
+
+Internally tile data is divided into “chunks.” Every tile is at least one chunk. Each tile has the following on-disk format:
+
+| **Field** | **Type** | **Description** |
+| :--- | :--- | :--- |
+| Num chunks | `uint64_t` | Number of chunks in the tile |
+| Chunk 1 | [Chunk](#chunk-format) | First chunk in the tile |
+| … | … | … |
+| Chunk N | [Chunk](#chunk-format) | N-th chunk in the tile |
+
+## Chunk Format 
+
+A chunk has the following on-disk format:
+
+| **Field** | **Type** | **Description** |
+| :--- | :--- | :--- |
+| Original length of chunk | `uint32_t` | The original \(unfiltered\) number of bytes of chunk data |
+| Filtered chunk length | `uint32_t` | The serialized \(filtered\) number of bytes of chunk data |
+| Chunk metadata length | `uint32_t` | Number of bytes in the chunk metadata |
+| Chunk metadata | `uint8_t[]` | Chunk metadata bytes |
+| Chunk filtered data | `uint8_t[]` | Filtered chunk bytes |
+
+The metadata added to a chunk depends on the sequence of filters in the pipeline used to filter the containing tile.
+
+If a pipeline used to filter tiles is empty \(contains no filters\), the tile is still divided into chunks and serialized according to the above format. In this case there are no chunk metadata bytes \(since there are no filters to add metadata\), and the filtered bytes are the same as original bytes.
+
+The “chunk metadata” before the actual "chunk filtered data" depend on the particular sequence of filters in the pipeline. In the simple case, each filter will simply concatenate its metadata to the chunk metadata region. Because some filters in the pipeline may wish to filter the metadata of previous filters \(e.g. compression, where it is beneficial to compress previous filters’ metadata in addition to the actual chunk data\), the ordering of filters also impacts the metadata that is eventually written to disk.
+
+The “chunk filtered data” bytes contain the final bytes of the chunk after being passed through the entire pipeline. When reading tiles from disk, these filtered bytes are passed through the filter pipeline in the reverse order.
```

Comment:
The bytes are not actually passed in reverse order right? I believe you mean to say that the `pipeline` is run in reverse?

Reply:
Yeap, that was there since the last version - I will fix.